DisguiseUnityRenderStream/Runtime/PipelineAbstraction/CameraCapture.cs

```diff
@@ -0,0 +1,275 @@
+                    m_copyDepth == other.m_copyDepth &&
+                    m_depthCopyFormat == other.m_depthCopyFormat &&
+                    m_depthCopyMode == other.m_depthCopyMode;
+            }
+            
+            public static bool operator ==(CameraCaptureDescription lhs, CameraCaptureDescription rhs) => lhs.Equals(rhs);
+
+            public static bool operator !=(CameraCaptureDescription lhs, CameraCaptureDescription rhs) => !(lhs == rhs);
+        }
+
+        /// <summary>
+        /// Called once textures have been captured and are ready to be consumed.
+        /// </summary>
+        public event Action<ScriptableRenderContext, CameraCapture> onTexturesReady = delegate {};
+        
+        /// <summary>
+        /// <para>
+        /// The captured color texture. Its format is defined by <see cref="description"/>.
+        /// </para>
+        ///
+        /// <para>
+        /// Note: while the texture may be configured for MSAA, Unity will always use an implicitly resolved version
+        /// of the texture for APIs such <see cref="CommandBuffer.CopyTexture(RenderTargetIdentifier, RenderTargetIdentifier)"/>
+        /// and see <see cref="Texture.GetNativeTexturePtr"/>.
+        /// </para>
+        ///
+        /// <remarks>
+        /// To avoid unnecessary texture copies, this refers directly to <see cref="Camera.targetTexture"/>.
+        /// </remarks>
+        /// </summary>
+        public RenderTexture cameraTexture => m_cameraTexture;
+        
+        /// <summary>
+        /// The captured depth texture. Its format is defined by <see cref="description"/>.
+        /// </summary>
+        public RenderTexture depthTexture => m_depthTexture;
+
+        /// <summary>
+        /// Defines the color and depth textures. The textures are automatically disposed and created on change.
+        /// </summary>
+        public CameraCaptureDescription description
+        {
+            get => m_description;
+            set
+            {
+                if (m_description != value)
+                {
+                    m_description = value;
+                    Refresh();
+                }
+            }
+        }
+
+        [SerializeField]
+        CameraCaptureDescription m_description = CameraCaptureDescription.Default;
+
+        DepthCopy m_depthCopy;
+        Camera m_camera;
+        RenderTexture m_cameraTexture;
+        RenderTexture m_depthTexture;
+        
+#if DEBUG
+        bool m_HasSetSecondNames;        
+#endif
+
+#if UNITY_EDITOR
+        CameraCaptureDescription m_lastDescription = CameraCaptureDescription.Default;
+        bool m_RefreshFlag; // Ensures thread safety (OnValidate is on a different thread)
+        
+        void OnValidate()
+        {
+            if (m_lastDescription != m_description)
+            {
+                m_lastDescription = m_description;
+                m_RefreshFlag = true;
+            }
+        }
+
+        void Update()
```

Comment:
The design was for Play mode (editor/player) only so far.

I tested in Edit mode and CameraCapture + DepthCopy work out of the box.

There's only issues with Presenter:
1. Presenter should skip preview and scene cameras by Camera.cameraType. Easy fix.
2. Presenter should blit into BuiltinRenderTextureType.CameraTarget (GameView RT) rather than null (the screen). This only makes a difference in Edit mode. Easy fix.
3. Only the Fill resize strategy works (this is the least useful strategy imo). The other strategies can't run correctly because `Screen.size` no longer refers to the Game View size in Edit mode, but to the Unity editor window screen size. The only way to fix it would be to Reflect into the internal GameView class. Not a massive hack but not great either.

I would defer Edit mode support for later because of the 3rd Presenter issue.

Reply:
Screen.size works well, it was a bug in my resize math. Fixed.

Using a WaitForEndOfFrame approach for Presenter in editor but exploring other solutions.

I like the low overhead in the Player approach.